### ISSUE TITLE
Nested dynamic content cache sections still broken

### ISSUE THREAD

**Issue by bwaidelich** [neos/neos-development-collection#1958]
### Description

This is a follow up to #1591 because the fix #1664 didn't solve all issues related to nested dynamic Content Cache sections apparently.

#### Steps to reproduce

Unfortunately I can't reliably reproduce the issue, but more people reported similar issues in Slack:
In a production setup I have a nested dynamic cache section like this:

- `cached`
  - `dynamic`
    - `cached`

With a dynamic `entryDiscriminator` that is dependent on some request arguments.

It all works as expected, but sometimes the dynamic content is rendered as

```
CONTENT_CACHE1e1ca04e3725be78b07b1d7cfe079ebd_74be16979710d4c4e7c6647856088456CONTENT_CACHE
```
And then for all following requests until cache is flushed.

#### Assumptions 

1. The `CACHE_PLACEHOLDER_REGEX` in https://github.com/neos/neos-development-collection/blob/master/Neos.Fusion/Classes/Core/Cache/ContentCache.php#L53 won't match the marker above because it contains an underscore. Shouldn't it be `[a-f0-9\_]+)` instead?
2. In this case we use a Redis Cache Backend – Could it be, that the cache segment tokens special characters are "lost in translation" due to some encoding issue? 

### Affected Versions

Neos: 3.3.8 (probably lower)

**Comment by bwaidelich** [comment]
Thanks to @kitsunet we found an easy way to reproduce the issue.
With the following Fusion:
```

root >
root = Neos.Fusion:Value {
    value = Neos.Fusion:Value {
        value = Neos.Fusion:Value {
            value = 'content'
            @cache {
                mode = 'dynamic'
                entryDiscriminator = 'static'
                context {
                    1 = 'node'
                }
                entryTags {
                    1 = 'testing'
                }
            }
        }

    }
    @cache {
        mode = 'cached'
    }
}
```
in place:

1. flush (content) caches
2. render the frontend
3. manually remove the outer ContentCache entry (the one that contains the `evalCached=` part)
4. re-render the frontend
5. refresh

#### Expected output (after refresh)

content

#### Actual output

CONTENT_CACHE<md5>_<md5>CONTENT_CACHE

**Comment by hlubek** [comment]
Could we put that scenario in a functional test to prevent regressions?

**Comment by bwaidelich** [comment]
@hlubek yes, that is the plan

**Comment by johannessteu** [comment]
Maybe related to #1165 as well and that use case could be tested against this pr

**Comment by hlubek** [comment]
Fixed by #1960
